internal/cli/atlas/deployments/search/indexes/delete.go

```diff
@@ -0,0 +1,154 @@
+	options.DeploymentOpts
+	search.IndexOpts
+	mongodbClient mongodbclient.MongoDBClient
+	store         store.SearchIndexDeleter
+}
+
+func (opts *DeleteOpts) Run(ctx context.Context) error {
+	if err := opts.validateAndPrompt(ctx); err != nil {
+		return err
+	}
+
+	if err := opts.Prompt(); err != nil {
+		return err
+	}
+
+	err := opts.RunLocal(ctx)
+	if err != nil && (errors.Is(err, options.ErrDeploymentNotFound)) {
+		return opts.RunAtlas()
+	}
+
+	return err
+}
+
+func (opts *DeleteOpts) RunAtlas() error {
+	return opts.Delete(opts.store.DeleteSearchIndex, opts.ConfigProjectID(), opts.DeploymentName)
+}
+
+func (opts *DeleteOpts) RunLocal(ctx context.Context) error {
+	connectionString, err := opts.ConnectionString(ctx)
+	if err != nil {
+		return err
+	}
+
+	if err = opts.mongodbClient.Connect(connectionString, connectWaitSeconds); err != nil {
+		return err
+	}
+	defer opts.mongodbClient.Disconnect()
+
+	return opts.Delete(opts.mongodbClient.DeleteSearchIndex, opts.Entry)
+}
+
+func (opts *DeleteOpts) initStore(ctx context.Context) func() error {
+	return func() error {
+		var err error
+		opts.store, err = store.New(store.AuthenticatedPreset(config.Default()), store.WithContext(ctx))
+		return err
+	}
+}
+
+func (opts *DeleteOpts) initMongoDBClient(ctx context.Context) func() error {
+	return func() error {
+		opts.mongodbClient = mongodbclient.NewClientWithContext(ctx)
+		return nil
+	}
+}
+
+func (opts *DeleteOpts) validateAndPrompt(ctx context.Context) error {
+	if opts.Entry == "" {
+		if err := promptRequiredName("Search Index ID", &opts.Entry); err != nil {
+			return err
+		}
+	}
+
+	if opts.DeploymentName == "" {
+		if err := opts.DeploymentOpts.Select(ctx); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func DeleteBuilder() *cobra.Command {
+	opts := &DeleteOpts{
+		DeleteOpts: cli.NewDeleteOpts(deletedMessage, deleteMessageFailed),
+	}
+	cmd := &cobra.Command{
+		Use:     "delete <indexId>",
+		Aliases: []string{"rm"},
+		Short:   "Delete the specified search index from the specified cluster.",
+		Long:    fmt.Sprintf(usage.RequiredRole, "Project Data Access Admin"),
+		Args:    require.MaximumNArgs(1),
+		Annotations: map[string]string{
+			"indexIdDesc": "ID of the index.",
+			"output":      opts.SuccessMessage(),
+		},
+		PreRunE: func(cmd *cobra.Command, args []string) error {
+			w := cmd.OutOrStdout()
+			opts.PodmanClient = podman.NewClient(log.IsDebugLevel(), w)
+			return opts.PreRunE(
+				opts.InitStore(opts.PodmanClient),
+				opts.initStore(cmd.Context()),
+				opts.initMongoDBClient(cmd.Context()),
+			)
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if len(args) > 0 {
+				opts.Entry = args[0]
+			}
+
+			return opts.Run(cmd.Context())
+		},
+	}
+
+	cmd.Flags().StringVar(&opts.DeploymentName, flag.DeploymentName, "", usage.ClusterName)
+	cmd.Flags().BoolVar(&opts.Confirm, flag.Force, false, usage.Force)
+
+	cmd.Flags().StringVar(&opts.ProjectID, flag.ProjectID, "", usage.ProjectID)
```

Comment:
[q] I'm wondering whether we need this flag across commands since it's also at the root command (non-blocking)

Reply:
I am not sure how users will be able to provide a different Projectid than the one in the default profile without this flag 🤔 what am I missing?

Reply:
we can check later anyway across all commands, my theory is that we inherit this flag from the root command "atlas" - this is not blocking anyways so I'll resolve